source/fe/mapping_q_generic.cc

```diff
@@ -1537,6 +1511,9 @@ namespace internal
             data.values_quad.resize(n_comp * n_q_points);
             data.gradients_quad.resize(n_comp * n_q_points * dim);
 
+            data.scratch.resize((dim - 1) *
+                                std::max(n_q_points, n_shape_values));
```

Comment:
This is the only part I don't understand. In `FEEvaluationImpl::evaluate` (which I think this can resolve to since the other evaluation kernels don't use the scratch array at all) we have the following:
```cpp
        temp1 = scratch_data;
        temp2 = temp1 +
                std::max(Utilities::fixed_power<dim>(shape_info.fe_degree + 1),
                         Utilities::fixed_power<dim>(shape_info.n_q_points_1d));
```
which implies that this array needs to be longer. Should it be longer?

Reply:
Technically the code should have been correct because we only need `temp2` in the 3D case. But this is hard to see and does not matter in terms of memory allocation, so I changed to `2 * std::max(...)` which fits both cases.